src/app/clusters/network-commissioning/network-commissioning.cpp

```diff
@@ -92,180 +96,406 @@ struct NetworkInfo
+{
+    return sClusterState;
+}
 
-            uint8_t extendedPanId[Thread::kSizeExtendedPanId];
+void SetCommissionerSecureSessionHandle(const SecureSessionHandle & secureSessionHandle)
+{
+    sCommissionerSecureSessionHandle = secureSessionHandle;
+}
 
-            static_assert(sizeof(sNetworks[i].mNetworkID) >= sizeof(extendedPanId),
-                          "Network ID must be larger than Thread extended PAN ID!");
-            SuccessOrExit(dataset.GetExtendedPanId(extendedPanId));
-            memcpy(sNetworks[i].mNetworkID, extendedPanId, sizeof(extendedPanId));
-            sNetworks[i].mNetworkIDLen = sizeof(extendedPanId);
+void EncodeAndSendEnableNetworkResponse(CHIP_ERROR err)
+{
+    CHIP_ERROR processError                  = CHIP_NO_ERROR;
+    chip::app::CommandSender * commandSender = nullptr;
+    EmberAfNetworkCommissioningError commissioningError =
+        (err == CHIP_NO_ERROR ? EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_SUCCESS
+                              : EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_UNKNOWN_ERROR);
+    NodeId commissionerNodeId              = sCommissionerSecureSessionHandle.GetPeerNodeId();
+    Transport::AdminId commissionerAdminId = sCommissionerSecureSessionHandle.GetAdminId();
+
+    // From the Network Commissioning cluster spec, we MAY put response command in seperate exchange, and may have a StatusCode
+    // before the ResponseCommand.
+    app::CommandPathParams cmdParams = { 0, /* group id */ 0, ZCL_NETWORK_COMMISSIONING_CLUSTER_ID,
+                                         ZCL_ENABLE_NETWORK_RESPONSE_COMMAND_ID, (chip::app::CommandPathFlags::kEndpointIdValid) };
+    TLV::TLVWriter * writer          = nullptr;
+    SuccessOrExit(processError = chip::app::InteractionModelEngine::GetInstance()->NewCommandSender(&commandSender));
+    SuccessOrExit(processError = commandSender->PrepareCommand(&cmdParams));
+    writer = commandSender->GetCommandDataElementTLVWriter();
+    SuccessOrExit(processError = writer->Put(TLV::ContextTag(0), static_cast<uint32_t>(commissioningError)));
+    SuccessOrExit(processError = writer->PutString(TLV::ContextTag(1), ""));
+    SuccessOrExit(processError = commandSender->FinishCommand());
+    SuccessOrExit(processError = commandSender->SendCommandRequest(commissionerNodeId, commissionerAdminId));
+
+    // Interaction model engine will manage the state of command sender when successfully send this command, so we can safely delete
+    // this.
+    commandSender = nullptr;
 
-            sNetworks[i].mNetworkType = NetworkType::kThread;
-            sNetworks[i].mEnabled     = false;
+exit:
+    if (commandSender != nullptr)
+    {
+        commandSender->Shutdown();
+    }
+    ChipLogError(Zcl, "Failed to send response command to commissioner: %d", processError);
+}
 
-            err = EMBER_ZCL_NETWORK_COMMISSIONING_ERROR_SUCCESS;
-            break;
+CHIP_ERROR __attribute__((warn_unused_result)) MoveClusterState(ClusterState newState)
```

Comment:
Please don't add these compiler specific extensions unconditionally. CHIP should be compatible with standard C++14.